src/Test.UnitTests.BinSkim.Driver/CompilerDataLoggerTests.cs

```diff
@@ -622,6 +625,135 @@ private void LogExceptionEvents(List<ExecutionException> exceptions, CompilerDat
             return sb;
         }
 
+        private StringBuilder ValidateChunkedEvents(string expectedEventName,
+                                                    List<ITelemetry> telemetryGeneratedEvents,
+                                                    string expectedChunkedContent,
+                                                    int chunkNumber)
+        {
+            var sb = new StringBuilder();
+            string assemblyEventsId = string.Empty;
+            string commandLineEventsId = string.Empty;
+
+            var compilerEvents = new List<EventTelemetry>();
+            var assemblyReferencesEvents = new List<EventTelemetry>();
+            var commandLineEvents = new List<EventTelemetry>();
+            var summaryEvents = new List<EventTelemetry>();
+
+            foreach (EventTelemetry telemetryEvent in telemetryGeneratedEvents)
+            {
+                switch (telemetryEvent.Name)
+                {
+                    case CompilerDataLogger.CompilerEventName:
+                        compilerEvents.Add(telemetryEvent);
+
+                        if (commandLineEventsId.Equals(string.Empty) && telemetryEvent.Properties.ContainsKey("commandLineId"))
+                        {
+                            commandLineEventsId = telemetryEvent.Properties["commandLineId"];
+                        }
+
+                        if (assemblyEventsId.Equals(string.Empty) && telemetryEvent.Properties.ContainsKey("assemblyReferencesId"))
+                        {
+                            assemblyEventsId = telemetryEvent.Properties["assemblyReferencesId"];
+                        }
+
+                        if (expectedEventName == CompilerDataLogger.AssemblyReferencesEventName && !telemetryEvent.Properties.ContainsKey("assemblyReferencesId"))
```

Comment:
>AssemblyReferencesEventName [](http://example.com/codeflow?start=68&length=27)

This line is getting very long, consider dropping your second condition in order to improve readability, here and elsewhere. I might actually prefer to do this everywhere to make the convention consistent.